plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3select/csv/S3SelectCsvRecordReader.java

```diff
@@ -94,27 +92,16 @@ public SelectObjectContentRequest buildSelectObjectRequest(Properties schema, St
         selectObjectCSVOutputSerialization.setQuoteCharacter(quoteChar);
         selectObjectCSVOutputSerialization.setQuoteEscapeCharacter(escapeChar);
         selectObjectOutputSerialization.setCsv(selectObjectCSVOutputSerialization);
-        selectObjectRequest.setOutputSerialization(selectObjectOutputSerialization);
-
-        // Scan range requests to S3 Select allow us to enable splits,
-        // by providing a finer granularity than file level.
-        // Works for CSV if AllowQuotedRecordDelimiter is disabled.
-        boolean isQuotedRecordDelimiterAllowed = Boolean.TRUE.equals(
-                selectObjectCSVInputSerialization.getAllowQuotedRecordDelimiter());
-        if (CompressionType.NONE.equals(compressionType) && !isQuotedRecordDelimiterAllowed) {
-            // Scan range queries are not supported for compressed files.
-            ScanRange scanRange = new ScanRange();
-            scanRange.setStart(getStart());
-            scanRange.setEnd(getEnd());
-            selectObjectRequest.setScanRange(scanRange);
-        }
 
-        return selectObjectRequest;
+        return selectObjectOutputSerialization;
     }
 
-    protected String getFieldDelimiter(Properties schema)
+    @Override
+    public boolean shouldEnableScanRange()
     {
-        // Use the field delimiter only if it is specified in the schema. If not, use a default field delimiter ','
-        return schema.getProperty(FIELD_DELIM, DEFAULT_FIELD_DELIMITER);
+        // Works for CSV if AllowQuotedRecordDelimiter is disabled.
+        boolean isQuotedRecordDelimiterAllowed = Boolean.TRUE.equals(
```

Comment:
Can we add tests for this scenario?

Reply:
Not really, currently, this field is not set on the InputSerialization (as this was the existent behaviour), but the check was added for completeness.